From 70e33753c7e836b20ecdfc775c0416c01bb98e8c Mon Sep 17 00:00:00 2001
From: HaHaWTH <id_cn00@outlook.com>
Date: Sun, 7 Apr 2024 16:43:25 +0800
Subject: [PATCH] prevent-IndexOutOfBounds


diff --git a/src/main/java/net/minecraft/server/Container.java b/src/main/java/net/minecraft/server/Container.java
index 3a488ce20..626a36dad 100644
--- a/src/main/java/net/minecraft/server/Container.java
+++ b/src/main/java/net/minecraft/server/Container.java
@@ -107,7 +107,13 @@ public abstract class Container {
     }
 
     public Slot getSlot(int i) {
-        return (Slot) this.slots.get(i);
+        // Beast start - Prevent IndexOutOfBoundsException
+        try {
+            return (Slot) this.slots.get(i);
+        } catch (IndexOutOfBoundsException e) {
+            return null;
+        }
+        // Beast end
     }
 
     public ItemStack shiftClick(EntityHuman entityhuman, int i) {
-- 
2.50.1.windows.1

