From 12b581519cbb11a235e171e8d3b056f6a80a8f75 Mon Sep 17 00:00:00 2001
From: ruViolence <78062896+ruViolence@users.noreply.github.com>
Date: Fri, 27 May 2022 21:22:05 +0500
Subject: [PATCH] fix: MC-128547

https://bugs.mojang.com/browse/MC-128547

diff --git a/src/main/java/net/minecraft/server/EnchantmentManager.java b/src/main/java/net/minecraft/server/EnchantmentManager.java
index f714dc326..84f849d5c 100644
--- a/src/main/java/net/minecraft/server/EnchantmentManager.java
+++ b/src/main/java/net/minecraft/server/EnchantmentManager.java
@@ -114,6 +114,7 @@ public class EnchantmentManager {
         EnchantmentManager.a.a = 0;
         EnchantmentManager.a.b = damagesource;
         a((EnchantmentManager.EnchantmentModifier) EnchantmentManager.a, iterable);
+        EnchantmentManager.a.b = null; // Reaper - Fix MC-128547
         return EnchantmentManager.a.a;
     }
 
@@ -141,6 +142,10 @@ public class EnchantmentManager {
             a((EnchantmentManager.EnchantmentModifier) EnchantmentManager.c, entityliving.getItemInMainHand());
         }
 
+        // Alr start - Fix MC-128547
+        EnchantmentManager.c.b = null;
+        EnchantmentManager.c.a = null;
+        // Alr end
     }
 
     public static void b(EntityLiving entityliving, Entity entity) {
@@ -154,6 +159,10 @@ public class EnchantmentManager {
             a((EnchantmentManager.EnchantmentModifier) EnchantmentManager.d, entityliving.getItemInMainHand());
         }
 
+        // Alr start - Fix MC-128547
+        EnchantmentManager.d.b = null;
+        EnchantmentManager.d.a = null;
+        // Alr end
     }
 
     public static int a(Enchantment enchantment, EntityLiving entityliving) {
-- 
2.50.1.windows.1

