From d72bc075afb5c01cdbfb74b2e879f5fcaa05e6a9 Mon Sep 17 00:00:00 2001
From: ruViolence <78062896+ruViolence@users.noreply.github.com>
Date: Fri, 3 Jun 2022 05:46:10 +0500
Subject: [PATCH] fix: do not load chunks on adjacent chest check


diff --git a/src/main/java/net/minecraft/server/TileEntityChest.java b/src/main/java/net/minecraft/server/TileEntityChest.java
index 631304122..f6d6a69a6 100644
--- a/src/main/java/net/minecraft/server/TileEntityChest.java
+++ b/src/main/java/net/minecraft/server/TileEntityChest.java
@@ -153,6 +153,7 @@ public class TileEntityChest extends TileEntityLootable { // Paper - Remove ITic
 
     public void o() {
         if (!this.a) {
+            if (this.world == null || !this.world.areChunksLoaded(this.position, 1)) return; // Reaper
             this.a = true;
             this.h = this.a(EnumDirection.WEST);
             this.g = this.a(EnumDirection.EAST);
-- 
2.50.1.windows.1

