From ea3eebb94e60240752ed64b5a5392f85435ea924 Mon Sep 17 00:00:00 2001
From: Adallei <Adallei@outlook.com>
Date: Wed, 22 Apr 2026 12:01:20 +0800
Subject: [PATCH] fix-player-dismount

Fixes a dupe:
https://youtu.be/U_u7nYokzSU(Donkey Dupe)

diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 9cc3d3421..73383ac1a 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -345,6 +345,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
 
                 // Paper start - Prevent moving into unloaded chunks
                 if (player.world.paperConfig.preventMovingIntoUnloadedChunks && !worldserver.isChunkLoaded((int) Math.floor(packetplayinvehiclemove.getX()) >> 4, (int) Math.floor(packetplayinvehiclemove.getZ()) >> 4, false)) {
+                    player.stopRiding(); // Alr - Force dismount
                     this.networkManager.sendPacket(new PacketPlayOutVehicleMove(entity));
                     return;
                 }
@@ -353,6 +354,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                 if (d10 - d9 > Math.max(100.0D, Math.pow((double) (org.spigotmc.SpigotConfig.movedTooQuicklyMultiplier * (float) i * speed), 2)) && (!this.minecraftServer.R() || !this.minecraftServer.Q().equals(entity.getName()))) { // Spigot
                 // CraftBukkit end
                     PlayerConnection.LOGGER.warn("{} (vehicle of {}) moved too quickly! {},{},{}", entity.getName(), this.player.getName(), Double.valueOf(d6), Double.valueOf(d7), Double.valueOf(d8));
+                    player.stopRiding(); // Alr - Force dismount
                     this.networkManager.sendPacket(new PacketPlayOutVehicleMove(entity));
                     return;
                 }
-- 
2.50.1.windows.1

