From c74608f13ac8b01c5f4a14bfb9ffaef238d11221 Mon Sep 17 00:00:00 2001
From: Adallei <Adallei@outlook.com>
Date: Thu, 11 Jun 2026 23:16:40 +0800
Subject: [PATCH] fix-shield-damage-sound


diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
index 7c04cbe87..000af5273 100644
--- a/src/main/java/net/minecraft/server/EntityLiving.java
+++ b/src/main/java/net/minecraft/server/EntityLiving.java
@@ -1433,6 +1433,7 @@ public abstract class EntityLiving extends Entity {
 
             // Apply blocking code // PAIL: steal from above
             if (event.getDamage(DamageModifier.BLOCKING) < 0) {
+                this.world.broadcastEntityEffect(this, (byte) 29); // SPIGOT-4635 - shield damage sound
                 this.damageShield((float) -event.getDamage(DamageModifier.BLOCKING));
                 Entity entity = damagesource.i();
 
-- 
2.50.1.windows.1

