From 506f95f045922890d9bea7b2965114eeb0fdf0d9 Mon Sep 17 00:00:00 2001
From: Adallei <Adallei@outlook.com>
Date: Tue, 16 Jun 2026 02:05:36 +0800
Subject: [PATCH] Paper-Update-entity-Metadata-for-all-tracked-players


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 3966e3632..a52a5844f 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1720,7 +1720,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
 
                     if (triggerLeashUpdate && (event.isCancelled() || this.player.inventory.getItemInHand() == null || this.player.inventory.getItemInHand().getItem() != Items.LEAD)) {
                         // Refresh the current leash state
-                        this.sendPacket(new PacketPlayOutAttachEntity(entity, ((EntityInsentient) entity).getLeashHolder()));
+                        entity.tracker.broadcast(new PacketPlayOutEntityMetadata(entity.getId(), entity.datawatcher, true)); // Paper - update entity for all players
                     }
 
                     if (event.isCancelled() || this.player.inventory.getItemInHand() == null || this.player.inventory.getItemInHand().getItem() != origItem) {
-- 
2.50.1.windows.1

